lms/models/application_instance.py

```diff
@@ -157,21 +157,23 @@ def update_lms_data(self, params: dict):
         """
         Update all the LMS-related attributes present in `params`.
 
-        If the current instance already has a `tool_consumer_instance_guid`
-        report it on logging and don't update any of the columns.
+        :raise ValueError: If tool_consumer_instance_guid is not present in params or the model itself.
+        :raise ReusedConsumerKey: If the tool_consumer_instance_guid present in params
+            is different than the one already recorded in the DB.
         """
 
         tool_consumer_instance_guid = params.get("tool_consumer_instance_guid")
-        if not tool_consumer_instance_guid:
-            # guid identifies the rest of the LMS data, if not there skip any updates
-            return
+        if not tool_consumer_instance_guid and not self.tool_consumer_instance_guid:
+            # We don't have a guid in either the request params or stored on the DB
+            raise ValueError("No GUID present in either request params or the DB")
```

Comment:
This is not a situation we've seen in any LMS yet so raising an unhanded exception will point us in the right direction if it ever happens.